presto-main/src/main/java/com/facebook/presto/statusDetector/StatusDetectorConfig.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.statusDetector;
+
+import com.facebook.airlift.configuration.Config;
+
+public class StatusDetectorConfig
+{
+    private boolean enabled;
+
+    public boolean isEnabled()
+    {
+        return enabled;
+    }
+
+    @Config("status-detector.enabled")
+    public StatusDetectorConfig setEnabled(boolean enabled)
+    {
+        this.enabled = enabled;
+        return this;
+    }
+}
```

Comment:
Given that we have the `no-op` implementation - do we need the enabled flag?

Reply:
Having a config is good, so if we want to disable it without code change, we can do it.

Reply:
using it in DiscoveryNodeManager to decide if to call the NodeStatusService or not. Can be useful in case we want to disable the feature without deployment.